### PR TITLE
Reviewing GAN basics, VisionDataModule, MNISTDataModule, CIFAR10DataModule

### PR DESCRIPTION
**Pull request by shivammehta007**
## What does this PR do?

Part of #839

## Before submitting

- [X] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure your PR does **only one thing**, instead of bundling different changes together?
- [ ] Did you make sure to **update the documentation** with your changes?
- [X] Did you write any **new necessary tests**? \[not needed for typos/docs\]
- [X] Did you verify new and **existing tests** pass locally with your changes?
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/lightning-bolts/blob/master/CHANGELOG.md)?

## PR review

- [ ] Is this pull request **ready for review**? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Review by otaj — COMMENTED** [review]
Hi, all in all, it looks good, thank you! :tada: However, I'd like to be sure that we are not raising any deprecation/user/any other warnings. For this, I just opened #844 which should then be used in tests (when it gets merged).

**Comment by shivammehta007** [comment]
Currently, all the modified classes and their test cases pass but with the exception that we filter `PossibleUserWarning` resulting in user suggestions as mentioned in https://github.com/Lightning-AI/lightning-bolts/pull/844#issuecomment-1196778171.

**Comment by otaj** [comment]
Awesome work! Let's wait for #844 to see if the tests are passing on CI as well, but otherwise it looks good!

**Review by otaj — APPROVED** [review]
LGTM! Let's wait for test and get this merged :+1: :muscle: :rocket:

File: pl_bolts/datamodules/vision_datamodule.py
Comment (luca-medeiros) [inline pl_bolts/datamodules/vision_datamodule.py:30]: Shouldn't we include these transforms into the class docstring?

Comment (shivammehta007) [reply pl_bolts/datamodules/vision_datamodule.py:30]: I agree we should add it. What will be an ideal way of doing this? Do I need to create another pull request for this?

Comment (otaj) [reply pl_bolts/datamodules/vision_datamodule.py:30]: I agree with this as well, good catch @luca-medeiros!

@shivammehta007, yes, opening a new PR is the only option (not even I have a push rights to master :joy:). Since this PR is already merged, there really is no other option. Just write there it's a followup of #843

Comment (luca-medeiros) [reply pl_bolts/datamodules/vision_datamodule.py:30]: @otaj 

Seems you are the only member doing reviews for #839. 
What about assembling a small (3~5 people) team to help you out with it? I would be willing to help review as well!

Comment (otaj) [reply pl_bolts/datamodules/vision_datamodule.py:30]: @luca-medeiros Oh, absolutely! That's the whole point of our Slack channel. I think you're not in there yet, can you ping me on PL Slack (`@Ota`) and I will add you there?

**Comment by shivammehta007** [comment]
As discussed adding docstring in a separate PR https://github.com/Lightning-AI/lightning-bolts/pull/847.

**Pull request merged**
